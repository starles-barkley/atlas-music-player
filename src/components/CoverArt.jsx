import Image from '../assets/placeholder.svg';

export default function CoverArt() {
    return (
        <img src={Image} alt="svg" />
    )
}