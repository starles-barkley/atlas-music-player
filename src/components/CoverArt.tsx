import Skeleton from 'react-loading-skeleton';
import 'react-loading-skeleton/dist/skeleton.css';
import { Song } from './MusicPlayer.tsx';

type coverArtProps = {
    loading: Boolean;
    playlist: Song[]
    currentSong: number
}
export function CoverArt({loading, playlist, currentSong}: coverArtProps) {
  const song = playlist[currentSong];

  return (
    <div className="relative w-full h-full">
      {loading ? (
        <Skeleton className="w-full h-90 rounded-lg" />
      ) : (
        <>
          <img src={song.cover} alt="Cover art" className="w-full h-full rounded-lg" />
          <div className="absolute inset-0 flex items-center justify-center bg-(--bg-color) bg-opacity-50 opacity-0 hover:opacity-100 transition-opacity duration-300 rounded-lg overflow-hidden">
            <div className="text-(--secondary) p-4 overflow-y-auto max-h-full">
              <p className="font-bold mb-2">Lyrics</p>
              <p>{song.lyrics}</p>
            </div>
          </div>
        </>
      )}
    </div>
  );
}