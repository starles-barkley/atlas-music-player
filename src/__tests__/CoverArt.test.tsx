import { render, cleanup } from "@testing-library/react";
import { expect, test, afterEach } from "vitest";
import CoverArt from "../components/CoverArt.tsx";


afterEach(() => {
    cleanup();
});


test("Cover Art image renders with valid cover image", () => {
    const { container } = render(<CoverArt cover="https://utfs.io/f/E9fJnaKtTy1bV09oPkISuh6fWpNsTRlAk1Qj9yqnVzCi32BL" />);
    expect(container).toMatchSnapshot();
});

test("Cover art renders with a different cover image", () => {
    const { container } = render(<CoverArt cover="https://utfs.io/f/E9fJnaKtTy1bdFkaBz4G9CJywD7oVn1hRLQAuEbf8xqkt34N" />);
    expect(container).toMatchSnapshot();
});