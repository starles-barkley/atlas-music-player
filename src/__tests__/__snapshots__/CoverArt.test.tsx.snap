// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cover Art image renders with valid cover image 1`] = `
<div>
  <div
    class="flex pl-6"
  >
    <img
      alt="song image"
      class="rounded-lg p-6 w-[600px] h-[500px]"
      src="https://utfs.io/f/E9fJnaKtTy1bV09oPkISuh6fWpNsTRlAk1Qj9yqnVzCi32BL"
    />
  </div>
</div>
`;

exports[`Cover art renders with a different cover image 1`] = `
<div>
  <div
    class="flex pl-6"
  >
    <img
      alt="song image"
      src="https://utfs.io/f/E9fJnaKtTy1bdFkaBz4G9CJywD7oVn1hRLQAuEbf8xqkt34N"
    />
  </div>
</div>
`;
